Galaxus.Functional.Tests/(Option)/OptionExtensions/OptionLinqExtensionsTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace Galaxus.Functional.Tests.OptionExtensions
+{
+    [TestFixture]
+    public class OptionLinqExtensionsTests
```

Comment:
The test method names should generally follow the METHOD_SCENARIO_RESULT-pattern. 

So for example: `ElementAtOrNone_IndexIsNegative_ReturnsNone()` or `ElementAtOrNone_IndexIsInBounds_ReturnsElementAtIndex()`. For your and my sanity, I won't comment that on every method :D